### ISSUE TITLE
NANs when pos_size or neg_size is zero

### ISSUE THREAD

**Issue by pfeatherstone** [valerystrizh/pytorch-histogram-loss#12]
You get NANs when pos_size or neg_size is zero.
It would also be helpful if you could give an example using randomly allocated tensors.
For example, to reproduce the NANs i use the following code but not sure if i'm using the loss function correctly:
```
t = torch.randn(2, 1536) #batch == 2, 1536 features
t = l2_normalisation(t)
labels = torch.zeros(2)
labels[0] = 1   #class 1
labels[1] = 34 #class 34
loss = HistogramLoss(256)
out = loss(t, labels)
```

**Comment by pfeatherstone** [comment]
@ii-research-yu not sure if @valerystrizh is maintaining this code anymore. Seeing as you've had a look at this repo before, do you mind giving your input. Thanks.
